astropy/time/core.py

```diff
@@ -803,12 +803,15 @@ def _apply(self, method=None, *args, **kwargs):
             # Apply the method to any value arrays (though skip if there is only
             # a single element and the method would return a view, since in
             # that case nothing would change).
-            if method is not None and val is not None:
-                if method == 'copy' or method == 'flatten' and val.size == 1:
+            if method is not None and val is not None and hasattr(val, method):
```

Comment:
Actually, `getattr` needs a string argument... (as does `hasattr`). I think I'll use this to correct something I didn't like that much anyway, that `method` was `None` for replication. It would seem much cleaner to just call with `method='replicate'`.